site/content/contribute/more-info/server/feature-flags.md

```diff
@@ -158,37 +177,22 @@ When the feature is rolled out to customers, logs will show if there are crashes
 
 ## Self-hosted releases
 
-For a feature-flagged feature to be included in a self-hosted release, the feature flag should be removed.
-
-Feature flags are generally off by default and self-hosted releases do not contact the management system. Therefore feature flags that are not ready for a self-hosted release will be automatically disabled for all self-hosted releases.
-
-Optionally the feature flag can be set to ``true`` in code if we are not yet ready to fully remove the feature flag. Some [examples are here](https://github.com/mattermost/mattermost-server/blob/master/model/feature_flags.go#L75).
+For a feature-flagged feature to be included in a self-hosted release, the feature flag should be either removed or set to ``true`` in code. The feature flag shouldn't be set to ``true`` in code if we're not yet ready to fully remove the feature flag. Some [examples are here](https://github.com/mattermost/mattermost-server/blob/master/model/feature_flags.go#L75).
 
 ## Tests
 
 Tests should be written to verify all states of the feature flag. Tests should cover any migrations that may take place in both directions (i.e., from "off" to "on" and from "on" to "off"). Ideally E2E tests should be written before the feature is merged, or at least before the feature flag is removed.
 
-# When to use
-
-There are no hard rules on when a feature flag should be used. It is left up to the best judgement of the responsible engineers to determine if a feature flag is required. The following are guidelines designed to help guide the determination.
-
-- Any "substantial" feature should have a flag
-- Features that are probably substantial:
-    - Features with new UI or changes to existing UI
-    - Features with a risk of regression
-- Features that are probably not substantial:
-    - Small bug fixes
-    - Refactoring
-    - Changes that are not user facing and can be completely verified by unit and E2E testing.
+Features that are not yet ready to be shipped in a self-hosted release are automatically disabled if the flag is set to ``false`` in code.
 
 ## Examples of feature flags
 
-< add some examples when we create them >
+Some [examples are here](https://github.com/mattermost/mattermost-server/blob/master/model/feature_flags.go#L75).
 
 ## FAQ
 
 1. What is the expected default value for boolean feature flags? Is it `true` or `false`?
-   - Definitely `false`. The idea is to use them to slowly roll out a feature. When the code is deployed, the feature flag is not enabled yet. See more details on feature flag rollout timelines [here](#timelines-for-rollouts).
+   - Initially `false`. The idea is to use them to slowly roll out a feature. When the code is deployed, the feature flag is not enabled yet. See more details on feature flag rollout timelines [here]({{< ref "/contribute/more-info/server/feature-flags#timelines-for-rollouts" >}}).
```

Comment:
I would agree that for cloud release this is correct: a feature flag should default to `false`. That gives us a week to vet that the feature doesn't break things.

However, for a self-hosted releases I think it's unreasonable to expect all feature flags to default to `false`.

Reply:
Clarified the text 👍